src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -109,5 +109,22 @@ public static void PrepareConstrainedRegionsNoOP()
         internal static bool IsPrimitiveType(this CorElementType et)
             // COR_ELEMENT_TYPE_I1,I2,I4,I8,U1,U2,U4,U8,R4,R8,I,U,CHAR,BOOLEAN
             => ((1 << (int)et) & 0b_0011_0000_0000_0011_1111_1111_1100) != 0;
+
+        /// <summary>Provide a fast way to access constant data stored in a module as a ReadOnlySpan{T}</summary>
+        /// <param name="fldHandle">A field handle that specifies the location of the data to be referred to by the ReadOnlySpan{T}. The Rva of the field must be aligned on a natural boundary of type T</param>
+        /// <returns>A ReadOnlySpan{T} of the data stored in the field</returns>
+        /// <exception cref="ArgumentException"><paramref name="fldHandle"/> does not refer to a field which is an Rva, is misaligned, or T is of an invalid type.</exception>
+        /// <remarks>This method is intended for compiler user rather than use directly in code. T must be one of byte, sbyte, char, short, ushort, int, long, ulong, float, or double.</remarks>
+        [Intrinsic]
+        public static ReadOnlySpan<T> CreateSpan<T>(RuntimeFieldHandle fldHandle)
+        {
+            unsafe
+            {
+                void* data = default;
+                int count = default;
+                GetSpanDataFrom(fldHandle, typeof(T).TypeHandle, &data, &count);
+                return new ReadOnlySpan<T>(data, count);
+            }
+        }
```

Comment:
```suggestion
        public static unsafe ReadOnlySpan<T> CreateSpan<T>(RuntimeFieldHandle fldHandle)
            => new ReadOnlySpan<T>(GetSpanDataFrom(fldHandle, typeof(T).TypeHandle, out int length), length);
```
Nit: You can make it one-liner if you change GetSpanDataFrom to return the pointer.